### ISSUE TITLE
False positive `svelte/no-unused-props` when object prop property is used via spread in script

### ISSUE THREAD

**Issue by Stadly** [sveltejs/eslint-plugin-svelte#1556]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.x

### What version of `eslint-plugin-svelte` are you using?

3.19.0

### What did you do?

```svelte
<!-- App.svelte -->
<script lang="ts">
  type Props = {
    app: { name: string; description: string };
  };
  const { app }: Props = $props();

  const data = { ...app, name: `${app.name} (copy)` };
</script>

<p>{data.name}</p>
```

### What did you expect to happen?

No error. `app.description` is used because `app` is spread into a new object — all properties of `app` (including `description`) flow into `data`.

### What actually happened?

```
'description' in 'app' is an unused property  svelte/no-unused-props
```

### Additional context

This is a different variant from the spread-into-component case fixed by #1378 (which handled `<Child {...props} />`). Here the prop is an object whose properties are consumed via a JavaScript object spread expression in `<script>`, not a component spread in the template.

The rule tracks which specific properties of an object-typed prop are accessed (e.g. `app.name`), but does not recognize that spreading the object (`{ ...app }`) constitutes usage of all its properties.
